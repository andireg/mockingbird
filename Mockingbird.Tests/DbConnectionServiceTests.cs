using Mockingbird.Tests.Services;
using Snapshooter.Xunit;
using System.Collections.Generic;
using System.Threading.Tasks;
using Xunit;

namespace Mockingbird.Tests
{
    public class DbConnectionServiceTests
    {
        [Fact]
        public async Task TestQuery()
        {
            using IMockContext<DbConnectionService> context = MockContextFactory.Start<DbConnectionService>();
            IEnumerable<DbConnectionItem> actual = await context.Instance.QueryAsync("KEY", default);
            Snapshot.Match(actual);
            context.Verify();
        }

        [Fact]
        public async Task TestExecuteScalar()
        {
            using IMockContext<DbConnectionService> context = MockContextFactory.Start<DbConnectionService>();
            long actual = await context.Instance.ExecuteScalarAsync("KEY", default);
            Assert.Equal(42, actual);
            context.Verify();
        }
    }
}