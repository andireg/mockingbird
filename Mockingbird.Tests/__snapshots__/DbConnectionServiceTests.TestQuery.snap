[
  {
    "Text": "FooBar",
    "Value": 13.24,
    "Date": "2021-01-01T00:00:00",
    "Boolean": true
  },
  {
    "Text": "FooBar2",
    "Value": null,
    "Date": "2022-01-01T00:00:00",
    "Boolean": false
  }
]
